src/node/in_memory_ext.rs

```diff
@@ -137,70 +142,61 @@ impl<S: ForkSource + std::fmt::Debug + Clone + Send + Sync + 'static> InMemoryNo
     /// `true` if a snapshot was reverted, otherwise `false`.
     pub fn revert_snapshot(&self, snapshot_id: U64) -> Result<bool> {
         let snapshots = self.snapshots.clone();
-        self.get_inner()
-            .write()
-            .map_err(|err| anyhow!("failed acquiring lock: {:?}", err))
-            .and_then(|mut writer| {
-                let mut snapshots = snapshots.write().map_err(|err| {
-                    anyhow!("failed acquiring read lock for snapshots: {:?}", err)
-                })?;
-                let snapshot_id_index = snapshot_id.as_usize().saturating_sub(1);
-                if snapshot_id_index >= snapshots.len() {
-                    return Err(anyhow!("no snapshot exists for the id '{}'", snapshot_id));
-                }
+        self.write_inner().and_then(|mut writer| {
+            let mut snapshots = snapshots
+                .write()
+                .map_err(|err| anyhow!("failed acquiring read lock for snapshots: {:?}", err))?;
+            let snapshot_id_index = snapshot_id.as_usize().saturating_sub(1);
+            if snapshot_id_index >= snapshots.len() {
+                return Err(anyhow!("no snapshot exists for the id '{}'", snapshot_id));
+            }
 
-                // remove all snapshots following the index and use the first snapshot for restore
-                let selected_snapshot = snapshots
-                    .drain(snapshot_id_index..)
-                    .next()
-                    .expect("unexpected failure, value must exist");
-
-                tracing::info!("Reverting node to snapshot '{snapshot_id:?}'");
-                writer
-                    .restore_snapshot(selected_snapshot)
-                    .map(|_| {
-                        tracing::info!("Reverting node to snapshot '{snapshot_id:?}'");
-                        true
-                    })
-                    .map_err(|err| anyhow!("{}", err))
-            })
+            // remove all snapshots following the index and use the first snapshot for restore
+            let selected_snapshot = snapshots
+                .drain(snapshot_id_index..)
+                .next()
+                .expect("unexpected failure, value must exist");
+
+            tracing::info!("Reverting node to snapshot '{snapshot_id:?}'");
```

Comment:
Throughout snapshot / mining modes we are using `tracing::info!`, do we want these to be written out by default? I know we want to move away from `tracing::info` all together but curious your thoughts on outputting this information.

Reply:
Yeah I am not adding new `tracing::info`, this line existed before this PR and I think we should refactor logging in a separate PR to reduce the noise.

Generally speaking IMO we should only use `tracing::trace` or `tracing::debug`.